sentry_streams/sentry_streams/pipeline/pipeline.py

```diff
@@ -136,6 +136,29 @@ def add(self, other: Pipeline) -> None:
         for source, dests in other.outgoing_edges.items():
             self.outgoing_edges[source] = dests
 
+    def start(self, step: Source | Branch) -> Pipeline:
```

Comment:
I'd still remove this method as you added streaming_source. This way we would remove the sequential coupling between start method and the following one.
Basically if we allow a Pipeline to be instantiated only with a state that is started, there will be less cognitive overhead for the users as they would not have to think about both started and not started states of the pipeline.
Moreover no start method makes it impossible to call it at the wrong time.
Let's jsut pass the source to the constructor ?

Overall, the more we can force the right behavior structurally, the better as people will have to learn less.